### PR TITLE
GDNative: fix StringName equal and less operators

### PR DESCRIPTION
**Pull request by Bromeon**
The `new` function accepts a pointer, where it constructs a `StringName` in-place:
```cpp
void GDAPI godot_string_name_new(godot_string_name *r_dest, const godot_string *p_name) {
	StringName *dest = (StringName *)r_dest;
	const String *name = (const String *)p_name;
	memnew_placement(dest, StringName(*name));
}
```
This means that it's possible to have _different pointers_ to the same `StringName`.

However, the `equal` and `less` methods only compare that pointer:
```cpp
godot_bool GDAPI godot_string_name_operator_equal(const godot_string_name *p_self, const godot_string_name *p_other) {
	const StringName *self = (const StringName *)p_self;
	const StringName *other = (const StringName *)p_other;
	return self == other;
}

godot_bool GDAPI godot_string_name_operator_less(const godot_string_name *p_self, const godot_string_name *p_other) {
	const StringName *self = (const StringName *)p_self;
	const StringName *other = (const StringName *)p_other;
	return self < other;
}
```

And to do that, they wouldn't need to first cast to `StringName*`. It looks like forgotten dereferencing.

### CODE REVIEW

**Comment by Bromeon** [comment]
I haven't had the chance to test this change in godot-rust yet, hopefully will get around to it later today.

**Review by akien-mga — APPROVED** [review]
Makes sense to me. I'll wait for feedback on how it fares in godot-rust to merge.

**Comment by Bromeon** [comment]
Added tests in https://github.com/godot-rust/godot-rust/pull/912, they are currently failing as expected.

When I compile Godot locally with the change in this PR, they pass. I couldn't use this exact revision since godot-rust is still based on Godot 3.4, so I cherry-picked my commit here onto an older Godot revision. 

TLDR: ready from my side.

**Comment by akien-mga** [comment]
Thanks!
